cedar-dafny/validation/subtyping.dfy

```diff
@@ -71,40 +75,46 @@ module validation.subtyping {
     }
   }
 
-  function lubEntity(lub1: EntityLUB, lub2: EntityLUB): EntityLUB {
-    lub1.union(lub2)
+  function lubEntity(lub1: EntityLUB, lub2: EntityLUB, m: ValidationMode): Result<EntityLUB> {
+    if m.isStrict() && lub1 != lub2
```

Comment:
In strict mode, do you also want to check that neither are `AnyEntity`?

Reply:
For strict mode this already gives `E </: AnyEntity` for `E` where `E != AnyEntity`. I could change it so that `AnyEntity </: AnyEntity`, but then subtyping isn't reflexive (which is strange). I don't think it really matters one way or the other since `AnyEntity` shouldn't be constructed by the strict typechecker as along as it doesn't show up in the request environment anywhere.